### PR TITLE
feat: add maintenance mode to runtime

### PR DESCRIPTION
**Pull request by palozano**
- chore: port from cumulus
- chore: last pass before correct install
- feat: configure mm pallet to disallow some calls
- chore: port maintenance to standalone

# Summary of changes

Add the functionality to enter the runtime into maintenance mode and
block the execution of some calls, while allowing the execution of
vital calls.

# Acceptance Checklist

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

## Dependencies

- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any teams which manage the dependencies have been notified of the breaking changes

## Bug (non-breaking change which fixes an issue)

- [ ] Have you **_written tests_** to protect against future occurrences of the bug?

## Feature (non-breaking change which adds functionality)

- [X] Have you **_seen it working_** in a development environment?
- [X] Have you **_written tests_** for **_happy_** and **_unhappy_** paths?
- [X] Have you implemented this feature as per the **_issue acceptance criteria_**?

### Substrate

If this change involves substrate, have you remembered:

- [ ] Storage Migration?
- [ ] RPC methods?
- [ ] Chainspec, both JSON specs & the module?
- [ ] Runtime versioning?
- [ ] Benchmarks?

## Breaking change (a feature that would cause existing functionality not to work as expected)

- [ ] Is the breaking change **_documented_**?
- [ ] Are your commits fully representative of the change?
- [ ] Has any previous functionality been **_deprecated_** and versioned?

## CI/CD

- [ ] If introducing new actions, have you **_looked at the action code_** for anything spurious?
- [ ] Have you written **_custom scripts_** for things that could be actions already on the marketplace?
- [ ] If introducing new actions, have you **_pegged a static version_**?

## Documentation Update

- [ ] Have you checked other documentation, such as the docs repository?
- [ ] If updating script documentation, have you **_tested it on both environments_**?

### CODE REVIEW

File: runtime/standalone/src/lib.rs [hunk 1]
```diff
@@ -124,5 +124,8 @@ construct_runtime!(
         PolkadotBridge: pallet_grandpa_finality_verifier::<Instance1> = 130,
         KusamaBridge: pallet_grandpa_finality_verifier::<Instance2> = 131,
 
+        // Handy utilities
+        MaintenanceMode: pallet_maintenance_mode = 132,
```
Comment (AwesomeIbex) [inline runtime/standalone/src/lib.rs:128]: This version would need to make room for light clients, so prolly want something like 140

File: runtime/standalone/src/system_config.rs [hunk 2]
```diff
@@ -185,3 +188,241 @@ impl pallet_utility::Config for Runtime {
...
+pub struct BaseCallFilter;
+impl Contains<Call> for BaseCallFilter {
+    fn contains(c: &Call) -> bool {
+        match c {
+            // System support
+            Call::System(_) => true,
+            // Call::ParachainSystem(_) => true,
+            Call::Timestamp(_) => true,
+            // Call::Preimage(_) => true,
+            // Call::Scheduler(_) => true,
```
Comment (AwesomeIbex) [inline runtime/standalone/src/system_config.rs:200]: Just remove anything to do with parachains here

File: runtime/standalone/src/system_config.rs [hunk 3]
```diff
@@ -185,3 +188,241 @@ impl pallet_utility::Config for Runtime {
...
+            // Collator support
+            Call::Authorship(_) => true,
+            // Call::CollatorSelection(_) => true,
+            // Call::Session(_) => true,
+            // XCM helpers
+            // Call::XcmpQueue(_) => true,
+            // Call::PolkadotXcm(_) => false,
+            // Call::DmpQueue(_) => true,
+            // Call::XBIPortal(_) => true,
+            // Call::AssetRegistry(_) => true,
```
Comment (AwesomeIbex) [inline runtime/standalone/src/system_config.rs:221]: same

File: runtime/standalone/src/lib.rs (hunk 1 above)
Comment (palozano) [reply runtime/standalone/src/lib.rs:128]: Done

File: runtime/standalone/src/system_config.rs (hunk 2 above)
Comment (palozano) [reply runtime/standalone/src/system_config.rs:200]: Done

File: runtime/standalone/src/system_config.rs (hunk 3 above)
Comment (palozano) [reply runtime/standalone/src/system_config.rs:221]: Done

File: runtime/standalone/src/system_config.rs [hunk 4]
```diff
@@ -1,19 +1,22 @@
...
-        ConstU32, NeverEnsureOrigin,
+        ConstU32, Contains, NeverEnsureOrigin, OffchainWorker, OnFinalize, OnIdle, OnInitialize,
+        OnRuntimeUpgrade,
     },
     PalletId,
 };
 use pallet_asset_tx_payment::HandleCredit;
 use polkadot_runtime_common::SlowAdjustingFeeUpdate;
-use sp_runtime::{traits::*, Permill};
+use sp_runtime::{traits::*, MultiAddress, Permill};
```
[the remark this answers is not in the document]
Comment (AwesomeIbex) [reply runtime/standalone/src/system_config.rs:19]: this needs to be fixed

[the remark this answers is not in the document]
Comment (palozano) [reply runtime/standalone/src/system_config.rs]: Removed

**Review by AwesomeIbex — DISMISSED** [review]

**Review by AwesomeIbex — DISMISSED** [review]

**Comment by 3h4x** [comment]
@AwesomeIbex you approval is not enough due to codeowners file
Screenshot 2023-04-28 at 11 31 40

**Comment by palozano** [comment]
While OOO: This PR was approved and can be merged.

**Review by MaciejBaj — APPROVED** [review]

**Review by MaciejBaj — APPROVED** [review]

**Review by MaciejBaj — APPROVED** [review]

**Pull request merged**